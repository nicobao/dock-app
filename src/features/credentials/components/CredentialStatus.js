import {NBox, Typography} from '../../../design-system';
import {HStack} from 'native-base';
import React from 'react';

import {useGetCredentialStatus} from '../hooks/credentialHooks';

export function CredentialStatus({credential}) {
  const {statusData} = useGetCredentialStatus({credential});

  return (
    <HStack ml={3}>
      <NBox mt={0.5}>{statusData.icon}</NBox>
      <Typography
        color={statusData.color}
        ml={1}
        variant={'credentialIssuanceDate'}>
        {statusData.message}
      </Typography>
    </HStack>
  );
}
