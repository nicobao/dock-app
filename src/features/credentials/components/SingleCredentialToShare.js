import React from 'react';
import {Checkbox} from 'native-base';

import {withErrorBoundary} from '../../../core/error-handler';
import {CredentialListItem} from '../CredentialsScreen';

export const SingleCredentialToShare = withErrorBoundary(
  ({rawCredential, isChecked, onSelect}) => {
    const credentialActions = (
      <Checkbox
        isChecked={isChecked}
        onChange={onSelect}
        accessibilityLabel="Select credential"
      />
    );
    return (
      <CredentialListItem
        credential={rawCredential.content}
        formattedData={rawCredential.formattedData}
        credentialActions={credentialActions}
      />
    );
  },
);
