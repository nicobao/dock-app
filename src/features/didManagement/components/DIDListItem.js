import {
  Button,
  DotsVerticalIcon,
  IconButton,
  NBox,
  Theme,
  Typography,
} from '../../../design-system';
import {HStack, VStack} from 'native-base';
import {addTestId} from '../../../core/automation-utils';
import {translate} from '../../../locales';
import React from 'react';

export function DIDListItem({didDocumentResolution, onOptionClicked, onShare}) {
  const {didDocument} = didDocumentResolution;
  return (
    <NBox
      my={3}
      py={3}
      pl={4}
      pr={3}
      style={{
        backgroundColor: Theme.colors.cardItemBackground,
        borderRadius: 12,
      }}>
      <VStack>
        <HStack>
          <Typography
            style={{
              flexGrow: 1,
            }}
            numberOfLines={1}
            pt={2}
            variant="didTitle">
            {typeof didDocumentResolution.name === 'string' &&
            didDocumentResolution.name.length > 0
              ? didDocumentResolution.name
              : translate('didManagement.no_name_set')}
          </Typography>
          <IconButton
            col
            {...addTestId('DIDListItemOptionButton')}
            onPress={onOptionClicked}>
            <DotsVerticalIcon />
          </IconButton>
        </HStack>
        <HStack>
          <Typography numberOfLines={1} mb={7} pt={2} variant="didDescription">
            {didDocument.id}
          </Typography>
        </HStack>
        <Button
          width="30%"
          size="xs"
          variant={'whiteButton'}
          colorScheme="dark"
          {...addTestId('Share')}
          onPress={onShare}>
          <Typography color={Theme.colors.primaryBackground}>
            {translate('didManagement.share')}
          </Typography>
        </Button>
      </VStack>
    </NBox>
  );
}
