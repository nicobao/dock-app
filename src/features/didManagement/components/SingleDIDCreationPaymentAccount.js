import {NBox, Typography} from '../../../design-system';
import React from 'react';
import {HStack, Stack} from 'native-base';
import {TokenAmount} from '../../tokens/ConfirmTransactionModal';
export function SingleDIDCreationPaymentAccount({item}) {
  return (
    <NBox>
      <HStack mb={1}>
        <Typography
          style={{
            flexGrow: 1,
          }}
          numberOfLines={1}
          textAlign="left"
          variant="description">
          {item.label}
        </Typography>
        <TokenAmount amount={item.balance}>
          {({tokenAmount, tokenSymbol}) => (
            <>
              <Stack direction="column">
                <Typography variant="description">
                  {tokenAmount} {tokenSymbol}
                </Typography>
              </Stack>
            </>
          )}
        </TokenAmount>
      </HStack>
      <Typography
        numberOfLines={1}
        textAlign="left"
        variant="screen-description">
        {item.description}
      </Typography>
    </NBox>
  );
}
