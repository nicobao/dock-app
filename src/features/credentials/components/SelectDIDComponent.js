import {NBox, Typography} from '../../../design-system';
import {FormControl} from 'native-base';
import {translate} from '../../../locales';
import {CustomSelectInput} from '../../../components/CustomSelectInput';
import {addTestId} from '../../../core/automation-utils';
import React from 'react';

export function SelectDIDComponent({dids, handleChange}) {
  return (
    <NBox>
      <FormControl>
        <Typography variant="h1" mb={12}>
          {translate('credentials.select_presentation_holder')}
        </Typography>

        <FormControl.Label>
          {translate('didManagement.did_name')}
        </FormControl.Label>
        <CustomSelectInput
          {...addTestId('DID')}
          onPressItem={item => {
            handleChange(item.value);
          }}
          renderItem={item => {
            return (
              <>
                <Typography
                  numberOfLines={1}
                  textAlign="left"
                  variant="description">
                  {item.label}
                </Typography>
                <Typography
                  numberOfLines={1}
                  textAlign="left"
                  variant="screen-description">
                  {item.description}
                </Typography>
              </>
            );
          }}
          items={dids}
        />
      </FormControl>
    </NBox>
  );
}
