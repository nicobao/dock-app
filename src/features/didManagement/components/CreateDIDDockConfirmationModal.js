import {Modal} from '../../../components/Modal';
import React from 'react';
import {translate} from '../../../locales';
import {LoadingButton, NBox, Typography} from '../../../design-system';
import {VStack} from 'native-base';
import {addTestId} from '../../../core/automation-utils';
import {AmountDetails} from '../../tokens/ConfirmTransactionModal';

export function CreateDIDDockConfirmationModal({
  didName,
  didType,
  onCreateDID,
  visible,
  onClose,
}) {
  const didTypeMap = {
    diddock: 'did:dock',
    didkey: 'did:key',
  };
  return (
    <Modal
      visible={visible}
      onClose={onClose}
      onBackButtonPress={onClose}
      modalSize={0.8}>
      <VStack p={7}>
        <Typography variant="h1" mb={4}>
          {translate('confirm_transaction.title')}
        </Typography>
        <NBox mt={5}>
          <Typography variant={'didModalInfoTitle'} mb={1}>
            {translate('didManagement.did_name')}
          </Typography>
          <Typography variant={'description'} mb={1}>
            {didName}
          </Typography>
        </NBox>

        <NBox mt={5}>
          <Typography variant={'didModalInfoTitle'} mb={1}>
            {translate('didManagement.did_type')}
          </Typography>
          <Typography variant={'description'} mb={1}>
            {didTypeMap[didType]}
          </Typography>
        </NBox>
        <NBox mt={5}>
          <Typography variant={'didModalInfoTitle'} mb={1}>
            {translate('confirm_transaction.fee')}
          </Typography>
          <AmountDetails amount={1} symbol={'DOCK'} />
        </NBox>
        <NBox mt={5}>
          <Typography variant={'didModalInfoTitle'} mb={1}>
            {translate('confirm_transaction.total')}
          </Typography>
          <AmountDetails amount={1} symbol={'DOCK'} />
        </NBox>
        <LoadingButton
          mt={12}
          full
          {...addTestId('CreateNewDIDScreenDIDCreate')}
          onPress={onCreateDID}>
          {translate('confirm_transaction.title')}
        </LoadingButton>
      </VStack>
    </Modal>
  );
}
