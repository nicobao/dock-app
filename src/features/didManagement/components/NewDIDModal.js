import {Modal} from '../../../components/Modal';
import React from 'react';
import {Stack} from 'native-base';
import {
  DownloadIcon,
  OptionList,
  PlusCircleIcon,
  Theme,
  Typography,
} from '../../../design-system';
import {translate} from '../../../locales';
import {addTestId} from '../../../core/automation-utils';
import {navigate} from '../../../core/navigation';
import {Routes} from '../../../core/routes';

export function NewDIDModal({visible, onClose, onImportDID}) {
  const content = (
    <Stack p={8} testID="addDIDModal">
      <Typography variant="h1">{translate('didManagement.add_did')}</Typography>
      <OptionList
        mt={5}
        items={[
          {
            testID: addTestId('CreateNewDID').testID,
            title: translate('didManagement.create_new_did'),
            icon: (
              <PlusCircleIcon
                style={{
                  color: Theme.colors.secondaryIconColor,
                }}
              />
            ),
            onPress: () => {
              onClose();
              navigate(Routes.DID_MANAGEMENT_NEW_DID);
            },
          },
          {
            testID: addTestId('ImportExistingDIDBtn').testID,
            title: translate('didManagement.import_existing_did'),
            icon: (
              <DownloadIcon
                style={{
                  color: Theme.colors.secondaryIconColor,
                }}
              />
            ),
            onPress: onImportDID,
          },
        ]}
      />
    </Stack>
  );
  return (
    <Modal visible={visible} onClose={onClose} onBackButtonPress={onClose}>
      {content}
    </Modal>
  );
}
