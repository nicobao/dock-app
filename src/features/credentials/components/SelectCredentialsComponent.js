import {NBox, Typography} from '../../../design-system';
import {translate} from '../../../locales';
import {SingleCredentialToShare} from './SingleCredentialToShare';
import {EmptyCredentials} from '../CredentialsScreen';
import React from 'react';
import {addTestId} from '../../../core/automation-utils';

export function SelectCredentialsComponent({
  credentials,
  selectedCredentials,
  setSelectedCredentials,
}) {
  return (
    <NBox>
      <Typography variant="h1" mb={2}>
        {translate('credentials.select_what_to_share')}
      </Typography>

      {credentials.length > 0 ? (
        credentials.map((item, index) => (
          <SingleCredentialToShare
            {...addTestId(`credentials_${index}`)}
            key={item.id}
            rawCredential={item}
            checked={selectedCredentials[item.id]}
            onSelect={isSelected => {
              setSelectedCredentials(prev => {
                return {
                  ...prev,
                  [item.id]: isSelected ? item.content : undefined,
                };
              });
            }}
          />
        ))
      ) : (
        <EmptyCredentials mt={'50%'} />
      )}
    </NBox>
  );
}
