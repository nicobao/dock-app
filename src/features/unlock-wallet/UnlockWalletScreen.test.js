import React from 'react';
import {shallow} from 'enzyme';
import configureMockStore from 'redux-mock-store';
import {UnlockWalletScreen} from './UnlockWalletScreen';
import {
  authenticationReducer,
  authenticationActions,
} from './unlock-wallet-slice';
import {NumericKeyboard} from '../../components/NumericKeyboard';

const mockStore = configureMockStore();

describe('UnlockWalletScreen', () => {
  it('should render correctly', () => {
    const initialState = {
      // placeholder for redux store
    };

    const wrapper = shallow(<UnlockWalletScreen biometry={true} />, {
      context: {store: mockStore(initialState)},
    });
    expect(wrapper.dive()).toMatchSnapshot();
  });
  it('should render keypad correctly', () => {
    const wrapper = shallow(
      <NumericKeyboard marginTop={50} onChange={jest.fn()} value={''} />,
      {},
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });
});

describe('Test Authentication', function () {
  it('should return the initial state', () => {
    expect(authenticationReducer(undefined, {})).toEqual({
      isLoggedIn: false,
    });
  });

  it('should handle auth state change', () => {
    const previousState = {
      isLoggedIn: false,
    };
    expect(
      authenticationReducer(
        previousState,
        authenticationActions.setAuth({isLoggedIn: true}),
      ),
    ).toEqual({
      isLoggedIn: true,
    });
  });
});
