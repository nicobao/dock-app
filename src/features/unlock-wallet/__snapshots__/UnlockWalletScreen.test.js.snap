// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnlockWalletScreen should render correctly 1`] = `
<RNCSafeAreaView
  accessibilityLabel="unlockWalletScreen"
  style={
    Object {
      "backgroundColor": "#141417",
      "flex": 1,
    }
  }
  testID="unlockWalletScreen"
>
  <Memo(ForwardRef(Box))
    flex={1}
  >
    <Memo(ForwardRef(Box))
      flex={1}
    >
      <Box
        justifyContent="center"
        row={true}
      >
        <Styled(Image)
          resizeMode="contain"
          source={
            Object {
              "testUri": "../../../src/assets/splash-logo.png",
            }
          }
          style={
            Object {
              "width": "57%",
            }
          }
        />
      </Box>
      <Styled(ScrollView)
        marginLeft={26}
        marginRight={26}
      >
        <PasscodeMask
          digits={6}
          filled={0}
          marginTop={20}
        />
        <Box
          alignItems="center"
        >
          <Styled(Component)
            marginTop={52}
            variant="h2"
          >
            Enter your passcode
          </Styled(Component)>
        </Box>
        <NumericKeyboard
          marginTop={50}
        />
        <Box
          alignItems="center"
          marginBottom={20}
        >
          <Styled(Component)
            variant="description"
          >
            Login with biometrics
          </Styled(Component)>
        </Box>
        <BuildIdentifier />
        <Modal>
          <Memo(ForwardRef(Stack))
            alignItems="center"
            m={5}
            p={5}
          >
            <Styled(Component)
              variant="h2"
            >
              App locked
            </Styled(Component)>
            <Styled(Component)
              variant="h4"
            >
              Please try again later
            </Styled(Component)>
            <Memo(ForwardRef(Stack))
              mt={4}
            >
              <Button>
                Close app
              </Button>
            </Memo(ForwardRef(Stack))>
          </Memo(ForwardRef(Stack))>
        </Modal>
      </Styled(ScrollView)>
    </Memo(ForwardRef(Box))>
  </Memo(ForwardRef(Box))>
</RNCSafeAreaView>
`;

exports[`UnlockWalletScreen should render keypad correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "marginTop": 50,
    }
  }
>
  <Box
    autoSize={true}
    flexDirection="row"
    justifyContent="center"
    marginBottom={24}
    row={true}
  >
    <KeyboardButton
      accessibilityLabel="keyboardNumber1"
      key="1"
      onPress={[Function]}
      testID="keyboardNumber1"
      value={1}
    />
    <KeyboardButton
      accessibilityLabel="keyboardNumber2"
      key="2"
      onPress={[Function]}
      testID="keyboardNumber2"
      value={2}
    />
    <KeyboardButton
      accessibilityLabel="keyboardNumber3"
      key="3"
      onPress={[Function]}
      testID="keyboardNumber3"
      value={3}
    />
  </Box>
  <Box
    autoSize={true}
    flexDirection="row"
    justifyContent="center"
    marginBottom={24}
    row={true}
  >
    <KeyboardButton
      accessibilityLabel="keyboardNumber4"
      key="4"
      onPress={[Function]}
      testID="keyboardNumber4"
      value={4}
    />
    <KeyboardButton
      accessibilityLabel="keyboardNumber5"
      key="5"
      onPress={[Function]}
      testID="keyboardNumber5"
      value={5}
    />
    <KeyboardButton
      accessibilityLabel="keyboardNumber6"
      key="6"
      onPress={[Function]}
      testID="keyboardNumber6"
      value={6}
    />
  </Box>
  <Box
    autoSize={true}
    flexDirection="row"
    justifyContent="center"
    marginBottom={24}
    row={true}
  >
    <KeyboardButton
      accessibilityLabel="keyboardNumber7"
      key="7"
      onPress={[Function]}
      testID="keyboardNumber7"
      value={7}
    />
    <KeyboardButton
      accessibilityLabel="keyboardNumber8"
      key="8"
      onPress={[Function]}
      testID="keyboardNumber8"
      value={8}
    />
    <KeyboardButton
      accessibilityLabel="keyboardNumber9"
      key="9"
      onPress={[Function]}
      testID="keyboardNumber9"
      value={9}
    />
  </Box>
  <Box
    autoSize={true}
    flexDirection="row"
    justifyContent="center"
    marginBottom={24}
    row={true}
  >
    <KeyboardButton
      accessibilityLabel="keyboardNumbernull"
      key="null"
      onPress={[Function]}
      testID="keyboardNumbernull"
      value={null}
    />
    <KeyboardButton
      accessibilityLabel="keyboardNumber0"
      key="0"
      onPress={[Function]}
      testID="keyboardNumber0"
      value={0}
    />
    <Box
      alignItems="center"
      flex={true}
      paddingTop={5}
    >
      <TouchableOpacity
        hitSlop={
          Object {
            "bottom": 10,
            "left": 10,
            "right": 10,
            "top": 10,
          }
        }
        onPress={[Function]}
      >
        <SvgFile />
      </TouchableOpacity>
    </Box>
  </Box>
</View>
`;
